pkg/chunk/table_manager.go

```diff
@@ -358,17 +386,29 @@ func (m *TableManager) partitionTables(ctx context.Context, descriptions []Table
 }
 
 func (m *TableManager) createTables(ctx context.Context, descriptions []TableDesc) error {
+	numFailures := 0
+	var firstErr error
+
 	for _, desc := range descriptions {
 		level.Info(util.Logger).Log("msg", "creating table", "table", desc.Name)
 		err := m.client.CreateTable(ctx, desc)
 		if err != nil {
-			return err
+			numFailures++
+			level.Error(util.Logger).Log("msg", "error creating table", "table", desc.Name, "err", err)
```

Comment:
This will lead to double logging (because the returned error is also logged in `loop()`). I understand you wanna to it because we will just return the first error, but I think a cleaner way to handle is:
1. Remove the log here
2. Import `tsdberrors "github.com/prometheus/prometheus/tsdb/errors"`  and use it:
  - `merr := tsdberrors.MultiError{}`
  - Then for each error, `merr.Add(err)`
  - At the end you can `return merr.Err()` (will return `nil` if no error has been registered)

What's your take?

_Same comment applies to `deleteTables()`_

Reply:
Ohh I didn't know we had something like that. Will fix it. Thanks!